File: examples/GoCZMQ/tasksink.go
//
//   Task sink
//   Binds PULL socket to tcp://localhost:5558
//   Collects results from workers via that socket
//   Initial Commit in examples/Go by Aaron Raddon
//   Ported to goczmq by Michael Guldan github.com/michaelcoyote
//   Requires: http://github.com/zeromq/goczmq
//
package main

import (
	"fmt"
	goczmq "github.com/zeromq/goczmq"
	"time"
)

func main() {

	//  Socket to receive messages on
	receiver := goczmq.NewSock(goczmq.Pull)
	defer receiver.Destroy()
	receiver.Bind("tcp://*:5558")

	//  Wait for start of batch
	msgbytes, _ := receiver.RecvMessage()
	fmt.Println("Received Start Msg ", string(msgbytes[0]))

	//  Start our clock now
	start_time := time.Now().UnixNano()

	//  Process 100 confirmations
	for i := 0; i < 100; i++ {
		msgbytes, _ = receiver.RecvMessage()
		fmt.Print(".")
	}

	//  Calculate and report duration of batch
	te := time.Now().UnixNano()
	fmt.Printf("Total elapsed time: %d msec\n", (te-start_time)/1e6)

}

File: examples/Free Pascal/tasksink.pas
{
  Task sink
  Binds PULL socket to tcp://localhost:5558
  Collects results from workers via that socket
  @author cpicanco <cpicanco@ufpa.br>
}
program tasksink;

{$MODE objfpc}{$H+}

uses SysUtils, zmq, zmq.helpers, zmq.types;

var
  context, receiver: Pointer;
  LString: string;
  start_time: LongWord;
  task_nbr: Integer;

begin
  //  Prepare our context and socket
  context := zmq_ctx_new;
  receiver := zmq_socket(context, ZMQ_PULL);
  zmq_bind(receiver, 'tcp://*:5558');

  //  Wait for start of batch
  LString := s_recv(receiver);
  LString := '';

  //  Start our clock now
  start_time := GettickCount64;

  //  Process 100 confirmations
  for task_nbr := 0 to 100 -1 do
  begin
    LString := s_recv(receiver);
    LString := '';
    if ((task_nbr / 10) * 10 = task_nbr) then
      Write(':')
    else
      Write('.');
    // Flush(output);
  end;

  //  Calculate and report duration of batch
  WriteLn(Format('Total elapsed time: %d msec',
    [GettickCount64 - start_time]));

  zmq_close(receiver);
  zmq_ctx_destroy(context);
end.

